ui/app/js/controllers/history_data_controller.js

```diff
@@ -0,0 +1,18 @@
+angular.module("app").controller("HistoryDataCtrl",
+function($scope, $http, $modalInstance, hs, hs_ct) {
+  $scope.hs = hs;
+  $scope.hs_ct = hs_ct;
+  $scope.rules_modal= false;
+  $scope.cancel = function () {
+    $modalInstance.dismiss('cancel');
+  };
+  // $scope.checkChangeType = function() {
```

Comment:
I think we should completely remove this code since we are not using it.

Reply:
Thanks @collin5 . I intend to use it later though. It is still a [WIP] but I'll clean up when I'm done